tests/test_start_date.py

```diff
@@ -0,0 +1,135 @@
+
+from tap_tester import connections, runner, LOGGER
+
+from base import IterableBase
+
+class StartDateTest(IterableBase):
+
+    # Creating variables to store two different start dates
+
+    start_date_1 = ""
+    start_date_2 = ""
+
+    def name(self):
+        return "tap_tester_iterable_start_date_test"
+
+    def test_run(self):
+        """
+            Test that the start_date configuration is respected
+            • verify that a sync with a later start date has at least one record synced
+            and less records than the 1st sync with a previous start date
+            • verify that each stream has less records than the earlier start date sync
+            • verify all data from later start data has bookmark values >= start_date
+            • verify that the minimum bookmark sent to the target for the later start_date sync
+            is greater than or equal to the start date
+            • verify by primary key values, that all records in the 1st sync are included in the 2nd sync.
+        """
+
+        self.start_date_1 = self.get_properties()
```

Comment:
Recently made changes to a few taps to swap start dates to make test more robust so that start date 1 is the more recent date and start date 2 is the older date.  This way if new data comes in between syncs the test does not break as sync 1 will always be a subset of sync 2 records.

Reply:
This tap has static test data so not sure we need to fix it here.

Reply:
Agreed.  Optional solution for dynamic data scenarios.

Reply:
Skipping this.